adaptive/utils.py

```diff
@@ -67,3 +68,20 @@ def decorator(method):
         return functools.wraps(other)(method)
 
     return decorator
+
+
+class _RequireAttrsABCMeta(abc.ABCMeta):
+    def __call__(self, *args, **kwargs):
+        obj = super().__call__(*args, **kwargs)
+        for name, type_ in obj.__annotations__.items():
+            try:
+                x = getattr(obj, name)
+            except AttributeError:
+                raise AttributeError(
+                    f"Required attribute {name} not set in __init__."
```

Comment:
What is clearer: "not set in __init__", "missing in the instance", "not set during instantiation"?